common/evaluators/pit2015_evaluator.py

```diff
@@ -3,32 +3,100 @@
+            fp += 1.0
+        elif guess == False and label == True:
+            fn += 1.0
+        elif guess == True and label == True:
+            tp += 1.0
+        elif guess == False and label == False:
+            tn += 1.0
+        if label == guess:
+            counter += 1.0
+
+    try:
+        P = tp / (tp + fp)
+        R = tp / (tp + fn)
+        F = 2 * P * R / (P + R)
+    except:
+        P = 0
+        R = 0
+        F = 0
+
+    accuracy = counter / len(predict_result)
+
+    maxF1 = 0
+    P_maxF1 = 0
+    R_maxF1 = 0
+    probs = predict_result
+    sortedindex = sorted(range(len(probs)), key=probs.__getitem__)
+    sortedindex.reverse()
+
+    truepos = 0
+    falsepos = 0
+    for sortedi in sortedindex:
+        if test_data_label[sortedi] == True:
+            truepos += 1
+        elif test_data_label[sortedi] == False:
+            falsepos += 1
+        precision = 0
+        if truepos + falsepos > 0:
+            precision = truepos / (truepos + falsepos)
+
+        if (tp + fn) > 0:
+            recall = truepos / (tp + fn)
+        else:
+            recall = 0
+        f1 = 0
+        if precision + recall > 0:
+            f1 = 2 * precision * recall / (precision + recall)
+            if f1 > maxF1:
+                # print probs[sortedi]
+                maxF1 = f1
+                P_maxF1 = precision
+                R_maxF1 = recall
+    # print("PRECISION: {}, RECALL: {}, max_F1: {}".format(P_maxF1, R_maxF1, maxF1))
+    return (accuracy, maxF1)
+
 class PIT2015Evaluator(Evaluator):
 
     def get_scores(self):
-        self.model.eval()
-        self.data_loader.init_epoch()
-        n_dev_correct = 0
-        total_loss = 0
-        acc_total = 0
-        rel_total = 0
-        pre_total = 0
-        for batch_idx, batch in enumerate(self.data_loader):
-            sent1, sent2 = self.get_sentence_embeddings(batch)
-            scores = self.model(sent1, sent2, batch.ext_feats, batch.dataset.word_to_doc_cnt, batch.sentence_1_raw, batch.sentence_2_raw)
-            prediction = torch.max(scores, 1)[1].view(batch.label.size()).data
-            gold_label = batch.label.data
-            n_dev_correct += (prediction == gold_label).sum().item()
-            acc_total += ((prediction == batch.label.data) * (prediction == 1)).sum().item()
-            total_loss += F.nll_loss(scores, batch.label, size_average=False).item()
-            rel_total += gold_label.sum().item()
-            pre_total += prediction.sum().item()
-
-            del scores
-
-        precision = acc_total / pre_total
-        recall = acc_total / rel_total
-        f1 = 2 * precision * recall / (precision + recall)
-        accuracy = 100. * n_dev_correct / len(self.data_loader.dataset.examples)
-        avg_loss = total_loss / len(self.data_loader.dataset.examples)
-        return [accuracy, avg_loss, precision, recall, f1], ['accuracy', 'cross_entropy_loss', 'precision', 'recall', 'f1']
+        #self.model.eval()
```

Comment:
Why comment this line?

Reply:
I use " torch.no_grad()" instead. 
" torch.no_grad()"  will reduce memory usage and speed up computations compared with "model.eval()".

Reply:
Interesting.